synapseclient/client.py

```diff
@@ -3853,47 +3853,104 @@ def _find_teams_for_principal(self, principal_id: str) -> typing.Iterator[Team]:
         for result in self._GET_paginated(f"/user/{principal_id}/team"):
             yield Team(**result)
 
-    @tracer.start_as_current_span("Synapse::getTeam")
-    def getTeam(self, id):
+    def create_team(
+        self,
+        name: str,
+        description: str = None,
+        icon: str = None,
+        opentelemetry_context=None,
+    ) -> Team:
+        """
+        Creates a new team.
+
+        Arguments:
+            name: The name of the team to create.
+            description: A description of the team.
```

Comment:
What was your rationale around these parameters?  A team object is: https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/Team.html.

Reply:
The reason was this on the Synapse API [docs](https://rest-docs.synapse.org/rest/POST/team.html).

> The passed request body may contain the following fields:
> 
> name - Give your new Team a name. The name must be unique, not used by an existing Team (required).
> description - a short text description of the Team's purpose (optional).
> icon - a fileHandle ID for an icon image file previously uploaded to Synapse (optional).

Reply:
The rest of the fields are filled in by the API for the old model, and then filled in by `fill_from_dict` in the new model

Reply:
Great point @thomasyu888 . @BWMac Could you make a modification here to pass along the possible creation parameters for the team for the REST API: https://rest-docs.synapse.org/rest/org/sagebionetworks/repo/model/Team.html

As Tom mentions I think it's just `canPublicJoin`, and `canRequestMembership`. The other fields look to be read only, not applicable to a create, or you're already passing it along. These two bools can be arguments to this method.

Reply:
added in my latest commit